File: api/BetterDaysAPI/Models/Usuario.cs
namespace BetterDaysAPI.Models
{
    public class Usuario 
    {
        public long idUsuario { get; set; }
        public string nome { get; set; } = "";
        public string loginUsuario { get; set; } = "";
        public string senhaUsuario { get; set; } = "";

        public virtual IEnumerable<Diario> Diarios { get; set; }
        public virtual IEnumerable<ListaMetas> ListaMetas { get; set; }
    }
}

File: lib/models/usuario.dart
class Usuario {
  final int? idUsuario;
  final String? nome;
  final String loginUsuario;
  final String senhaUsuario;

  Usuario(
      {required this.loginUsuario,
      required this.senhaUsuario,
      this.idUsuario,
      this.nome});

  @override
  String toString() =>
      'Usuário {idUsuario: $idUsuario, nome: $nome, loginUsuario: $loginUsuario, senhaUsuario: $senhaUsuario}';

  // Transforma um JSON na classe e captura seus valores
  Usuario.fromJson(Map<String, dynamic> json)
      : idUsuario = json['idUsuario'],
        nome = json['nome'],
        loginUsuario = json['loginUsuario'],
        senhaUsuario = '';

  // Transforma a classe em um JSON
  Map<String, dynamic> toJson() => {
        'nome': nome,
        'loginUsuario': loginUsuario,
        'senhaUsuario': senhaUsuario,
      };
}
